src/definitions/x509/trust_anchor.rs

```diff
@@ -0,0 +1,317 @@
+    Aamva(X509),
+    Custom(X509, ValidationRuleSet),
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct ValidationRuleSet {
+    pub distinguished_names: Vec<String>,
+    #[serde(rename = "type")]
+    pub typ: RuleSetType,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub enum RuleSetType {
+    IACA,
+    AAMVA,
+    Custom,
+    ReaderAuth,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct TrustAnchorRegistry {
+    pub certificates: Vec<TrustAnchor>,
+}
+
+pub fn process_validation_outcomes(
+    leaf_certificate: CertificateInner,
+    root_certificate: CertificateInner,
+    rule_set: ValidationRuleSet,
+) -> Vec<X509Error> {
+    let mut errors: Vec<X509Error> = vec![];
+
+    //execute checks on x509 components
+    match apply_ruleset(leaf_certificate, root_certificate.clone(), rule_set) {
+        Ok(mut v) => {
+            errors.append(&mut v);
+        }
+        Err(e) => {
+            errors.push(e);
+        }
+    }
+
+    // make sure that the trust anchor is still valid
+    errors.append(&mut check_validity_period(&root_certificate));
+
+    //TODO: check CRL to make sure the certificates have not been revoked
+    errors
+}
+
+pub fn validate_with_ruleset(
+    leaf_certificate: CertificateInner,
+    trust_anchor: TrustAnchor,
+) -> Vec<X509Error> {
+    let mut errors: Vec<X509Error> = vec![];
+
+    match trust_anchor {
+        TrustAnchor::Iaca(certificate) => {
+            let rule_set = ValidationRuleSet {
+                distinguished_names: vec!["2.5.4.6".to_string(), "2.5.4.8".to_string()],
+                typ: RuleSetType::IACA,
+            };
+            match x509_cert::Certificate::from_der(&certificate.bytes) {
+                Ok(root_certificate) => {
+                    errors.append(&mut process_validation_outcomes(
+                        leaf_certificate,
+                        root_certificate,
+                        rule_set,
+                    ));
+                }
+                Err(e) => {
+                    errors.push(e.into());
+                }
+            };
+        }
+        TrustAnchor::Aamva(certificate) => {
+            let rule_set = ValidationRuleSet {
+                distinguished_names: vec!["2.5.4.6".to_string(), "2.5.4.8".to_string()],
+                typ: RuleSetType::AAMVA,
+            };
+            //The Aamva ruleset follows the IACA ruleset, but makes the ST value mandatory
+            match x509_cert::Certificate::from_der(&certificate.bytes) {
+                Ok(root_certificate) => {
+                    errors.append(&mut process_validation_outcomes(
+                        leaf_certificate,
+                        root_certificate,
+                        rule_set,
+                    ));
+                }
+                Err(e) => {
+                    errors.push(e.into());
+                }
+            };
+        }
+        TrustAnchor::Custom(_certificate, _ruleset) => {
+            //TODO
+        }
```

Comment:
Since this isn't supported, can we just remove the `Custom` TrustAnchor enum variant for now?